packages/remix-dev/config/format.ts

```diff
@@ -13,11 +13,22 @@ export function formatRoutes(
   routeManifest: RouteManifest,
   format: RoutesFormat
 ) {
+  let routes = createHierarchicalRoutes<ConfigRoute, JsonFormattedRoute>(
```

Comment:
Create a hierarchy once and then print from the same structure with the right format